From: =?utf-8?q?J=C3=BCrg_Billeter?= <j@bitron.ch>
Date: Tue, 28 Nov 2023 20:04:17 +0100
Subject: playlist-manager: Fix crash with libxml2 2.12
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

The missing include resulted in an implicit function declaration, which
may result in a segmentation fault on startup.

    ../shell/rb-playlist-manager.c: In function ‘rb_playlist_manager_load_playlists’:
    ../shell/rb-playlist-manager.c:374:23: warning: implicit declaration of function ‘xmlParseMemory’ [-Wimplicit-function-declaration]
      374 |                 doc = xmlParseMemory (g_bytes_get_data (data, NULL), g_bytes_get_size (data));
          |                       ^~~~~~~~~~~~~~
    ../shell/rb-playlist-manager.c:374:23: warning: nested extern declaration of ‘xmlParseMemory’ [-Wnested-externs]
    ../shell/rb-playlist-manager.c:374:21: warning: assignment to ‘xmlDocPtr’ {aka ‘struct _xmlDoc *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
      374 |                 doc = xmlParseMemory (g_bytes_get_data (data, NULL), g_bytes_get_size (data));
          |                     ^

(cherry picked from commit 66da2f843e84bfa85870a4ace4054aa91ea07951)

Origin: upstream, after 3.4.7
---
 shell/rb-playlist-manager.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/shell/rb-playlist-manager.c b/shell/rb-playlist-manager.c
index 4974c9f..e0a54c6 100644
--- a/shell/rb-playlist-manager.c
+++ b/shell/rb-playlist-manager.c
@@ -41,6 +41,7 @@
 #include <stdio.h>      /* rename() */
 #include <unistd.h>     /* unlink() */
 
+#include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
